moz-webgpu-cts/src/main.rs

```diff
@@ -125,6 +190,474 @@ fn run(cli: Cli) -> ExitCode {
+
+            let (exec_reports_sender, exec_reports_receiver) = channel();
+            exec_report_paths
+                .into_par_iter()
+                .for_each_with(exec_reports_sender, |sender, path| {
+                    let res = fs::File::open(&path)
+                        .map(BufReader::new)
+                        .map_err(Report::msg)
+                        .wrap_err("failed to open file")
+                        .and_then(|reader| {
+                            serde_json::from_reader::<_, ExecutionReport>(reader)
+                                .into_diagnostic()
+                                .wrap_err("failed to parse JSON")
+                        })
+                        .wrap_err_with(|| {
+                            format!(
+                                "failed to read WPT execution report from {}",
+                                path.display()
+                            )
+                        })
+                        .map(|parsed| (path, parsed))
+                        .map_err(|e| {
+                            log::error!("{e:?}");
+                            AlreadyReportedToCommandline
+                        });
+                    let _ = sender.send(res);
+                });
+
+            for res in exec_reports_receiver {
+                let (_path, exec_report) = match res {
+                    Ok(ok) => ok,
+                    Err(AlreadyReportedToCommandline) => return ExitCode::FAILURE,
+                };
+
+                let ExecutionReport {
+                    run_info:
+                        RunInfo {
+                            platform,
+                            build_profile,
+                        },
+                    entries,
+                } = exec_report;
+
+                for entry in entries {
+                    let TestExecutionEntry { test_name, result } = entry;
+
+                    let test_path = TestPath::from_execution_report(&test_name).unwrap();
+                    // TODO: ignore things outside of the WPT path(s) we care about
+                    let TestOutcomes {
+                        test_outcomes: recorded_test_outcomes,
+                        subtests: recorded_subtests,
+                    } = outcomes_by_test
+                        .entry(test_path.clone().into_owned())
+                        .or_default();
+
+                    let (reported_outcome, reported_subtests) = match result {
+                        TestExecutionResult::Complete {
+                            duration: _,
+                            outcome,
+                            subtests,
+                        } => (outcome, subtests),
+                        TestExecutionResult::JobMaybeTimedOut { status, subtests } => {
+                            if !status.is_empty() {
+                                // TODO: probably don't keep this
```

Comment:
I think following up on this can be combined with deserializing logic; see <https://github.com/ErichDonGubler/moz-webgpu-cts/pull/14/files#r1380588610>.

Reply:
https://github.com/ErichDonGubler/moz-webgpu-cts/issues/49, 0820bd8